sktime/performance_metrics/tests/test_performance_metrics_forecasting.py

```diff
@@ -2,64 +2,524 @@
+    },
+    "median_squared_percentage_error": {
+        "test_case_1": 0.029589708748632582,
+        "test_case_2": 0.023172298452886965,
+        "test_case_3": 1.0,
+        "func": median_squared_percentage_error,
+        "class": MedianSquaredPercentageError(),
+    },
+    "root_mean_squared_percentage_error": {
+        "test_case_1": 0.17898108940463758,
+        "test_case_2": 0.18513472990780716,
+        "test_case_3": 2251799813685248.0,
+        "func": mean_squared_percentage_error,
+        "class": MeanSquaredPercentageError(square_root=True),
+    },
+    "root_median_squared_percentage_error": {
+        "test_case_1": 0.17201659439900727,
+        "test_case_2": 0.15222450017289255,
+        "test_case_3": 1.0,
+        "func": median_squared_percentage_error,
+        "class": MedianSquaredPercentageError(square_root=True),
+    },
+    "mean_relative_absolute_error": {
+        "test_case_1": 0.485695805,
+        "test_case_2": 0.477896036,
+        "test_case_3": 0.875,
+        "func": mean_relative_absolute_error,
+        "class": MeanRelativeAbsoluteError(),
+    },
+    "median_relative_absolute_error": {
+        "test_case_1": 0.411364556,
+        "test_case_2": 0.453437859,
+        "test_case_3": 1.0,
+        "func": median_relative_absolute_error,
+        "class": MedianRelativeAbsoluteError(),
+    },
+    "geometric_mean_relative_absolute_error": {
+        "test_case_1": 0.363521894,
+        "test_case_2": 0.402438951,
+        "test_case_3": 3.6839e-07,
+        "func": geometric_mean_relative_absolute_error,
+        "class": GeometricMeanRelativeAbsoluteError(),
+    },
+    "geometric_mean_relative_squared_error": {
+        "test_case_1": 0.132148167,
+        "test_case_2": 0.161957109,
+        "test_case_3": 4.517843023201426e-07,
+        "func": geometric_mean_relative_squared_error,
+        "class": GeometricMeanRelativeSquaredError(),
+    },
+}
 
 
-@pytest.mark.parametrize("test_y", TEST_YS_ZERO)
-def test_mape_loss_near_zero(test_y):
-    # mape should be large when predicitons are close to zero
-    y_test = test_y
-    y_pred = y_test + 0.001
-    assert mape_loss(y_test, y_pred) > 1e6
+@pytest.mark.parametrize("loss_func_name", LOSS_RESULTS.keys())
+@pytest.mark.parametrize("n_test_case", [1, 2, 3])
+def test_univariate_loss_zero_error(n_test_case, loss_func_name):
+    loss_class = LOSS_RESULTS[loss_func_name]["class"]
+    loss_func = LOSS_RESULTS[loss_func_name]["func"]
+    true_loss = 0
+    y_true = Y_TEST_CASES[f"test_case_{n_test_case}"]["test"]
+    y_train = Y_TEST_CASES[f"test_case_{n_test_case}"]["train"]
+    # Testing case of perfect forecaster
+    y_pred = y_true
+    # Also perfect benchmark
+    y_pred_benchmark = y_true
 
+    loss_func_args = inspect.getfullargspec(loss_func).args
+    if "y_train" in loss_func_args:
+        function_loss = loss_func(y_true, y_pred, y_train)
+        class_loss = loss_class.fn(y_true, y_pred, y_train)
```

Comment:
shouldn't we here test whether `__call__` works as expected rather than `fn` attribute?

Reply:
I was also thinking whether we should make `fn` private and rename it to `_func`, more readable in my opinion, and avoids users using it rather than going through `__call__`

Reply:
That does seem better. I'll update that. Right now we can't test the call because the call doesn't accept y_train (I think our plan was to complete this pull request based on #712, and follow-up right away with a second on that to update the classes with that). 

Note that the second pull request will also make the test function part much easier (no need to conditionally calculate loss, etc).